app/controllers/collaborators/base_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class Collaborators::BaseController < Sellers::BaseController
+  before_action :authorize_user
+
+  layout "inertia"
+
+  inertia_share { CollaboratorPresenter.new(seller: current_seller).inertia_shared_props }
+
+  def destroy
+    raise NotImplementedError, "Subclasses must implement destroy" unless block_given?
+
+    @collaborator.mark_deleted!
+
+    if current_seller == @collaborator.seller
+      AffiliateMailer.collaboration_ended_by_seller(@collaborator.id).deliver_later
+    elsif current_seller == @collaborator.affiliate_user
+      AffiliateMailer.collaboration_ended_by_affiliate_user(@collaborator.id).deliver_later
+    end
+
+    yield if block_given?
+  end
```

Comment:
This action is used on `/collaborators` and `/collaborators/incomings` page. 

The yielded block at the end will let the inheriting class define it's own routing logic for re-direction. Checkout `incomings_controller` and `main_controller` for examples